doc/src/manual/metaprogramming.md

```diff
@@ -126,8 +126,18 @@ julia> Symbol(:var,'_',"sym")
 :var_sym
 ```
 
-Note that to use `:` syntax, the symbol's name must be a valid identifier.
-Otherwise the `Symbol(str)` constructor must be used.
+To create a symbol that isn't a valid identifier, the `Symbol(str)` constructor must be used:
+
+```jldoctest
+julia> Symbol(1)
+Symbol("1")
+
+julia> Symbol(true)
+Symbol("true")
```

Comment:
Wouldn't it be better to instead have examples of `Symbol(::String)` (I guess that is what you meant with `Symbol(str)`)? At least update one of those examples so it doesn't look like `Symbol(x)` would give `Symbol("x")`.

Reply:
`str` was just what it said originally and wasn't sure about changing it. I don't have a strong opinion.